src/mechanics/elasticity.cc

```diff
@@ -398,34 +480,40 @@ void Elasticity::update_output_fields()
 void Elasticity::zero_time_step()
 {
 	START_TIMER(EqData::name());
-	data_.mark_input_times( *time_ );
-	data_.set_time(time_->step(), LimitSide::right);
+	eq_fields_->mark_input_times( *time_ );
+	eq_fields_->set_time(time_->step(), LimitSide::right);
 	std::stringstream ss; // print warning message with table of uninitialized fields
 	if ( FieldCommon::print_message_table(ss, "mechanics") ) {
 		WarningOut() << ss.str();
 	}
 
-    ( (LinSys_PETSC *)ls )->set_initial_guess_nonzero();
+    ( (LinSys_PETSC *)eq_data_->ls )->set_initial_guess_nonzero();
```

Comment:
If possible move construction and initialization of the LinSys into single method avoiding typecasts.